### PR TITLE
chore: moving e2e tests to TypeScript, including typings for protractor.

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

**Comment by hansl** [comment]
FYI @jelbourn  @robertmesserle

**Comment by hansl** [comment]
The change to forbidden-identifiers is to not block on something like `exit` because it doesn't check if `xit` is preceded/succeeded by a non word token.

File: e2e/typings/es6-shim/es6-shim.d.ts [hunk 1]
```diff
@@ -0,0 +1,670 @@
+// Compiled using typings@0.6.10
```
Comment (jelbourn) [inline e2e/typings/es6-shim/es6-shim.d.ts]: Why do we have to copy these typings into our project instead of installing them?

**Comment by jelbourn** [comment]
Looks like the changes to the forbidden keywords script is broken on CI

**Comment by hansl** [comment]
I'm going to move the forbidden keyword script into a separate PR. Still using this PR for testing though

**Comment by jelbourn** [comment]
LGTM

**Pull request closed without merging**